src/main/java/teammates/storage/api/CoursesDb.java

```diff
@@ -153,4 +165,14 @@ private Course getCourseEntity(String courseId) {
     
         return courseList.get(0);
     }
+    
+    private List<Course> getCourseEntities(List<String> courseIds) {
+        Query q = getPM().newQuery(Course.class);
+        q.setFilter(":p.contains(ID)");
+
+        @SuppressWarnings("unchecked")
+        List<Course> courses = (List<Course>) q.execute(courseIds);
+
+        return courses;
+    }
```

Comment:
Might not need to change this, I think this still makes the same number of queries as the original?